### DISCUSSION TITLE
installation process

### DISCUSSION THREAD

**Discussion by scorpio1379-1** [duhow/xiaoai-patch#32] [Q&A]
good day. i have some questions about installation process for LX06 speaker. i've managed to backup boot0,data0 and system0 partition as @danielk117 described in /research/LX06/install.md  those files have *.dump file name , but when i execute sudo make extract FILE=image-mtd4 after docker run -it -v $PWD:/xiaoai xiaoai-patch, i receive "No such file " error. i understand that i missed image creation step , but how can i make correct image?  @danielk117 just say "create a img file using binwalk" but how can i make it? should i download lx06/mico_firmware_9c712_1.74.10.bin or maybe system0.dump (it looks like system0 is rootfs, but i m not sure) , how i must "flash the not-in-use rootfs partition" ? can you give some more detailed instruction?

**Comment by duhow** [comment]
There are many ways to gather the system image.
Whatever you choose, use common sense to know what is the parameter you have to provide.

So, if you did backup `system0` and that image contains the system base you want to use, then run
```bash
sudo make extract FILE=system0-image-filename
```

You can check what is the **in-use rootfs** partition by using different commands:
```bash
df -h
ls -l /dev/root
```

The **non-in-use** will be the partition that is not mentioned there. 🙂

**Reply by scorpio1379-1** [reply]
i must execute df -h on speaker os , right? but how can can i log in , assuming that i have only micro-usb connection to speaker as @danielk117 described in install.md?

**Comment by danielk117** [comment]
> just say "create a img file using binwalk" but how can i make it? should i download lx06/mico_firmware_9c712_1.74.10.bin or maybe system0.dump

Both should create an img file, `binwalk -e system0.dump` or `binwalk -e mico_firmware_9c712_1.74.10.bin`.

**Reply by scorpio1379-1** [reply]
`binwalk -e system0.dump` create a directory with ./_system0.dump.extracted/squashfs-root dir and 0.squashfs file
 I assume that it do same as `sudo make extract FILE=system0.dump`
Am I right?

**Comment by scorpio1379-1** [comment]
just to make all things clearer: 
I made backup by `update.exe mread store system0 normal 0x40000000 system0.dump`
then 
```
sudo make clean
sudo make extract FILE=system0.dump
sudo make patch MODEL=lx06
```

then in ./squashfs-root/etc/shadow I have change root password how it described in  install.md

then 
`sudo make build MODEL=lx06`

in ./release/ I will get image file to flash (say it will be new_lx06_image)

```
update.exe partition boot1 boot0.dump
update.exe partition system1 new_lx06_image.img

```
do not understand this one though 

- flash `system0` partition
  - if you want to patch the installed firmware on your speaker (the factory installed version might be newer than the one we try to flash),
  then skip flashing the `system0` and set `/usr/bin/fw_env -s boot_part boot1`,
  follow the next step and then get an copy of `mtd4`, i.e. by `dd if=/dev/mtd4 of=/tmp/system0.img` and `scp` for transfering to your computer


I have to flash both partitions(system0 and system1) with same image?
How can I execute 
 `/usr/bin/fw_env -s boot_part boot1` or
`dd if=/dev/mtd4 of=/tmp/system0.img`
if I only have usb connection and can do something only by update.exe?

**Reply by danielk117** [reply]
> if you want to patch the installed firmware on your speaker (the factory installed version might be newer than the one we try to flash)

you can patch a dumped or downloaded firmware (its your choice). for me, the dumped firmware was never than latest available downloadable firmware, so I wanted to patch the dump instead.

at this step in the instruction, you should already haved flashed `boot0`, `boot1` and `system1`. flash `system0` too or follow the last optional step in the manual (https://github.com/duhow/xiaoai-patch/blob/master/research/lx06/install.md) -> i updated it to makes things clear 😉
